plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -695,8 +695,11 @@ public Optional<ConnectorOutputMetadata> finishCreateTable(ConnectorSession sess
     public Optional<ConnectorTableLayout> getInsertLayout(ConnectorSession session, ConnectorTableHandle tableHandle)
     {
         IcebergTableHandle table = (IcebergTableHandle) tableHandle;
-        Table icebergTable = catalog.loadTable(session, table.getSchemaTableName());
-        return getWriteLayout(icebergTable.schema(), icebergTable.spec(), false);
+        Schema schema = SchemaParser.fromJson(table.getTableSchemaJson());
```

Comment:
`SchemaParser.fromJson` isn't very expensive because it's cached.
If it was not cached, it could be expensive.

`catalog.loadTable` isn't very expensive because it's cached.
if it was not cached, we could have query consistency issues (like one query accessing same table twice, and reading different versions)

how do we assess which one is actually better?
it sounds that in any case we rely on some (hidden) caching taking place.

Reply:
Personally, I find `SchemaParser.fromJson(tableHandle.getSchemaJson())` less hidden than the `loadTable` option

Reply:
Okay, but don't we rely on caching `catalog.loadTable` anyway?
e.g. we want to read consistent table version in self-join query case.
So this is not only about schemas (which we can and do carry in the table handle), but also about other state.

I think i tend to agree that -- if we carry info in a table handle, this is "the version of information" to be used